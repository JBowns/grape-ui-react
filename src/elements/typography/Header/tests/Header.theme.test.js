import React from 'react';
import { ThemeProvider } from 'styled-components';
import renderer from 'react-test-renderer';
import { Header } from '..';
import 'jest-styled-components';

describe('Header using Theme color', () => {
  it('should return a Header object with custom color', () => {
    const theme = {
      colors: {
        grapeSoda: {
          base: 'hsl(325, 84.6%, 28%)',
          dark: 'hsl(305, 33.9%, 23.7%)',
          light: 'hsl(313, 67.8%, 47.5%)',
        },
      },
    };
    const element = (
      <ThemeProvider theme={theme}>
        <Header bg="grapeSoda.dark" color="grapeSoda.light">Lorem Ipsum</Header>
      </ThemeProvider>
    );
    const component = renderer.create(element);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Header using Theme Font Family', () => {
  it('should return a Header object with a custom font family', () => {
    const theme = {
      fonts: {
        trueSpace: '"DejaVu Sans Mono", "Bitstream Vera Sans Mono", Courier',
      },
    };
    const element = (
      <ThemeProvider theme={theme}>
        <Header fontFamily="trueSpace">Lorem Ipsum</Header>
      </ThemeProvider>
    );
    const component = renderer.create(element);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Header using overridden base Font Family', () => {
  it('should return a Header object with the custom base font family', () => {
    const theme = {
      fonts: {
        base: 'Frutiger, "Frutiger Linotype", Univers, Calibri, "Gill Sans", "Gill Sans MT", "Myriad Pro", Myriad, "DejaVu Sans Condensed", "Liberation Sans", "Nimbus Sans L", Tahoma, Geneva, "Helvetica Neue", Helvetica, Arial, sans-serif',
      },
    };
    const element = (
      <ThemeProvider theme={theme}>
        <Header.h2>Lorem Ipsum</Header.h2>
      </ThemeProvider>
    );
    const component = renderer.create(element);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
