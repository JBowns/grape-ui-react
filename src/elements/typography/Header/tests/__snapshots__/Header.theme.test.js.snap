// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header using Theme Font Family should return a Header object with a custom font family 1`] = `
.c0 {
  background-color: transparent;
  color: inherit;
  font-family: "DejaVu Sans Mono","Bitstream Vera Sans Mono",Courier;
  font-size: 2.5rem;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin: 0 0 1rem;
}

<h1
  className="c0"
>
  Lorem Ipsum
</h1>
`;

exports[`Header using Theme color should return a Header object with custom color 1`] = `
.c0 {
  background-color: hsl(305,33.9%,23.7%);
  color: hsl(313,67.8%,47.5%);
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 2.5rem;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin: 0 0 1rem;
}

<h1
  className="c0"
>
  Lorem Ipsum
</h1>
`;

exports[`Header using overridden base Font Family should return a Header object with the custom base font family 1`] = `
.c0 {
  background-color: transparent;
  color: inherit;
  font-family: Frutiger,"Frutiger Linotype",Univers,Calibri,"Gill Sans","Gill Sans MT","Myriad Pro",Myriad,"DejaVu Sans Condensed","Liberation Sans","Nimbus Sans L",Tahoma,Geneva,"Helvetica Neue",Helvetica,Arial,sans-serif;
  font-size: 2rem;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin: 0 0 1rem;
}

<h2
  className="c0"
>
  Lorem Ipsum
</h2>
`;
